tsfresh/feature_extraction/feature_calculators.py

```diff
@@ -2212,3 +2214,67 @@ def benford_correlation(x):
     # np.corrcoef outputs the normalized covariance (correlation) between benford_distribution and data_distribution.
     # In this case returns a 2x2 matrix, the  [0, 1] and [1, 1] are the values between the two arrays
     return np.corrcoef(benford_distribution, data_distribution)[0, 1]
+
+
+@set_property("fctype", "combiner")
+def matrix_profile(x, param):
+    """
+    TODO: Documentation
+
+    :param x: the time series to calculate the feature of
+    :type x: numpy.ndarray
+    :param param: contains dictionaries {"sample_pct": x, "threshold": y, "feature": z}
```

Comment:
Computing the PMP by default can be a bit expensive in some cases as many MatrixProfiles are computed. Are you OK with having a window parameter instead of a threshold here? Another approach could be to always default to the maximum window size given a user threshold to determine what MatrixProfile window to compute:

https://matrixprofile.docs.matrixprofile.org/api.html#matrixprofile-algorithms-maximum-subsequence

To determine the maximum "subsequence" (window size), we still have to compute MatrixProfiles to determine what makes sense. It will probably make a little more sense if you take a look at that code. The algorithm starts with a very wide search and refines the window size later on.

Reply:
I'd recommend the default behavior (that is, no user-defined window is passed) be computation of the maximum subsequence with the current 0.95 correlation threshold. I realize that the threshold can be very domain-specific, but this seems like the best compromise for an "automated" solution that many users will prefer to leverage.

@nils-braun to your point, more advanced users can specify window size(s) by passing variables directly to `compute`

Reply:
The other item that could be confusing for users is that the sample_pct parameter uses the "anytime" algorithm (approximate solution) under the covers. This may result in different answers, however a faster solution in some cases.